Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1418,37 +1311,122 @@ bool RenderTheme::hasListButtonPressed(const RenderObject& renderer) const
 }
 #endif
 
-#if PLATFORM(IOS_FAMILY)
+// FIXME: iOS does not use this so arguably this should be better abstracted. Or maybe we should
+// investigate if we can bring the various ports closer together.
+void RenderTheme::adjustButtonOrCheckboxOrColorWellOrInnerSpinButtonOrRadioOrSwitchStyle(RenderStyle& style, const Element* element) const
+{
+    auto autoAppearance = autoAppearanceForElement(style, element);
+    auto appearance = adjustAppearanceForElement(style, element, autoAppearance);
```

Comment:
I think we can just use `style.effectiveAppearance()` rather than recompute it?

Reply:
If that's equivalent I'm happy to do that. I wasn't sure.

Reply:
Yes. In fact, the current logic is incorrect since it would override additional adjustments made to the effective appearance in `RenderTheme::adjustStyle`.